xline-client/examples/error_handling.rs

```diff
@@ -0,0 +1,43 @@
+//! An example to show how the errors are organized in `xline-client`
+use curp::error::CommandProposeError;
+use xline::storage::ExecuteError;
+use xline_client::{
+    error::{ClientError, Result},
+    types::kv::PutRequest,
+    Client, ClientOptions,
+};
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    // the name and address of all curp members
+    let curp_members = [
+        ("server0", "10.0.0.1:2379"),
+        ("server1", "10.0.0.2:2379"),
+        ("server2", "10.0.0.3:2379"),
+    ];
+
+    let client = Client::connect(curp_members, ClientOptions::default())
+        .await?
+        .kv_client();
+
+    // We try to update the key using its previous value.
+    // It should return an error and it should be `key not found`
+    // as we did not add it before.
+    let resp = client
+        .put(PutRequest::new("key", "").with_ignore_value(true))
+        .await;
+    let err = resp.unwrap_err();
+
+    // We first match the client error
+    let ClientError::ProposeError(pe) = err else { unreachable!("should be a propose error")};
```

Comment:
Error message should be "XXXX should be err after compaction". XXXX stands for some actions.